From 2ec7750ef7d4c6a443b0fb04b1627efabb9d7c03 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Tue, 2 Dec 2014 10:34:35 +0200
Subject: [PATCH] init: fix splash fifo, and make config file optional

just having the fbsplash.ppm in boot media activates now splash.
---
 initramfs-init.in | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/initramfs-init.in b/initramfs-init.in
index ecab5dd..6357965 100755
--- a/initramfs-init.in
+++ b/initramfs-init.in
@@ -677,9 +677,17 @@ if [ -f "$sysroot/etc/.default_boot_services" -o ! -f "$ovl" ]; then
 	rm -f "$sysroot/etc/.default_boot_services"
 fi
 
-if [ "$KOPT_splash" != "no" -a -e $ALPINE_MNT/fbsplash.ppm -a -e $ALPINE_MNT/fbsplash.cfg ]; then
+if [ "$KOPT_splash" != "no" -a -e $ALPINE_MNT/fbsplash.ppm ]; then
+	local config
 	ebegin "Starting bootsplash (from $ALPINE_MNT)"
-	setsid fbsplash -T 16 -s $ALPINE_MNT/fbsplash.ppm -i $ALPINE_MNT/fbsplash.cfg -f $sysroot/$splashfile &
+	mkfifo $sysroot/$splashfile
+	if [ -e $ALPINE_MNT/fbsplash.cfg ]; then
+		config=$ALPINE_MNT/fbsplash.cfg
+	else
+		config=/tmp/fbsplash.cfg
+		echo "IMAGE_ALIGN=CM" > $config
+	fi
+	setsid fbsplash -T 16 -s $ALPINE_MNT/fbsplash.ppm -i $config -f $sysroot/$splashfile &
 	eend 0
 else
 	KOPT_splash="no"
-- 
2.2.0

